plugins/in_dbus/in_dbus.c

```diff
@@ -0,0 +1,527 @@
+
+    reply = dbus_message_new_method_return(msg);
+
+    dbus_message_iter_init_append(reply, &args);
+    if (!dbus_message_iter_append_basic(
+                &args, DBUS_TYPE_STRING, &description)) {
+        flb_error("[in_dbus] Could not append description");
+    }
+    else if (!dbus_connection_send(conn, reply, &serial)) {
+        flb_error("[in_dbus] Could not send reply");
+    }
+    else {
+        dbus_connection_flush(conn);
+    }
+    dbus_message_unref(reply);
+}
+
+static void in_dbus_log_dict(struct flb_in_dbus_config *dbus_config,
+                             struct flb_time *out_time,
+                             DBusMessage *msg,
+                             DBusMessageIter *args,
+                             DBusConnection *conn)
+{
+    DBusMessageIter dict;
+    DBusMessage* reply;
+    dbus_uint32_t serial = 0;
+
+    /*  Make sure that we're at the right part of the message */
+    if (DBUS_TYPE_ARRAY != dbus_message_iter_get_arg_type(args)) {
+        flb_error("[in_dbus] Message is not a dictionary");
+        in_dbus_reply_error(msg, conn, "Method call has invalid arguments");
+        return;
+    }
+
+    /*  Write the time for this sample */
+    msgpack_pack_array(&dbus_config->mp_pck, 2);
+    flb_time_append_to_msgpack(out_time, &dbus_config->mp_pck, 0);
+
+    /*  Count up arguments, and record this in the pack */
+    unsigned count = 0;
+    for (dbus_message_iter_recurse(args, &dict);
+         dbus_message_iter_get_arg_type(&dict) != DBUS_TYPE_INVALID;
+         dbus_message_iter_next(&dict), count++);
+    msgpack_pack_map(&dbus_config->mp_pck, count);
+
+    /*  Iterate over items in the map, packing each one */
+    for (dbus_message_iter_recurse(args, &dict);
+         dbus_message_iter_get_arg_type(&dict) != DBUS_TYPE_INVALID;
+         dbus_message_iter_next(&dict))
+    {
+        DBusMessageIter entry;
+        DBusMessageIter variant;
+        char* key = "";
+
+        if (dbus_message_iter_get_arg_type(&dict) != DBUS_TYPE_DICT_ENTRY) {
+            flb_error("[in_dbus] Expected dictionary key");
+            break;
+        }
+
+        dbus_message_iter_recurse(&dict, &entry);
+        if (DBUS_TYPE_STRING != dbus_message_iter_get_arg_type(&entry)) {
+            flb_error("[in_dbus] Expected string key");
+            break;
+        }
+        dbus_message_iter_get_basic(&entry, &key);
+
+        dbus_message_iter_next(&entry);
+        if (DBUS_TYPE_VARIANT != dbus_message_iter_get_arg_type(&entry)) {
+            flb_error("[in_dbus] Expected variant key");
+            break;
+        }
+
+        dbus_message_iter_recurse(&entry, &variant);
+
+        {   /* Pack the key */
+            size_t key_len = strlen(key);
+            msgpack_pack_str(&dbus_config->mp_pck, key_len);
+            msgpack_pack_str_body(&dbus_config->mp_pck, key, key_len);
+        }
+
+        switch (dbus_message_iter_get_arg_type(&variant)) {
+#define GET(DBUS_TYPE, TYPE, MSGPACK_FUNC)                  \
```

Comment:
Maybe remove the `{ }` and the terminating `;` on the `break`? This will make `GET()` without a terminating `;` a syntax error. You'll need to move out the variable declaration, but those are forbidden in the style guide anyway.

Reply:
Although this `GET` macro isn't actually parsed as a statement, so we don't need to require a semicolon.